src/twisted/internet/test/test_iocp.py

```diff
@@ -210,3 +226,39 @@ def cb(self, rc, bytes, evt):
         self.assertEqual(fd.counter, EVENTS_PER_LOOP)
         ir.doIteration(0)
         self.assertEqual(fd.counter, EVENTS_PER_LOOP + 1)
+
+    def test_doubleHalfClose(self):
+        """
+        If one side half-closes its connection, and then the other side of the
+        connection calls C{loseWriteConnection}, and then C{loseConnection} in
+        {writeConnectionLost}, the connection is closed correctly.
+
+        This covers a code branch for ticket #9553.
```

Comment:
Can you make this an actual hyperlink (epytext syntax `U{https://...}` ?  Eventually we're going to have to migrate bugtrackers, and it'll be nice to just have hyperlinks that the server can redirect rather than have to preserve the numbering scheme.